evap/rewards/forms.py

```diff
@@ -1,14 +1,108 @@
+from contextlib import contextmanager
+from datetime import date
+
 from django import forms
+from django.core.exceptions import ValidationError
+from django.core.validators import MaxValueValidator, StepValueValidator
+from django.db import transaction
+from django.utils.translation import gettext as _
 
-from evap.rewards.models import RewardPointRedemptionEvent
+from evap.evaluation.models import UserProfile
+from evap.rewards.models import RewardPointRedemption, RewardPointRedemptionEvent
+from evap.rewards.tools import reward_points_of_user
 
 
 class RewardPointRedemptionEventForm(forms.ModelForm):
     class Meta:
         model = RewardPointRedemptionEvent
-        fields = ("name", "date", "redeem_end_date")
+        fields = ("name", "date", "redeem_end_date", "step")
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.fields["date"].localize = True
         self.fields["redeem_end_date"].localize = True
+
+
+class RewardPointRedemptionForm(forms.Form):
+    event = forms.ModelChoiceField(queryset=RewardPointRedemptionEvent.objects.all(), widget=forms.HiddenInput())
+    points = forms.IntegerField(min_value=0, label="")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not self.initial:
+            return
+
+        self.fields["points"].validators.append(MaxValueValidator(self.initial["total_points_available"]))
+        self.fields["points"].widget.attrs["max"] = self.initial["total_points_available"]
```

Comment:
@hansegucker where does "max" come from, and why do we need a separate `MaxValueValidator`? In the [django docs on IntegerField](https://docs.djangoproject.com/en/5.1/ref/forms/fields/#django.forms.IntegerField), it sounds to me as if the field automatically does min- and max value validation if the values are set up, but it says that this should be done as `min_value` and `max_value`. We use `min_value` above, why not just `max_value` here?

Same for the StepValidator below, the IntegerField is supposed to have support for `step_size`.

Reply:
We have to differentiate on form fields and form widgets. max is a form widget attribute, max_value is a form field attribute. At this time, the form is already initialized and changing max_value doesn't have any effect anymore.

Reply:
Can we just do
```python3
self.fields['year'] = forms.IntegerField(
    min_value=0,
    max_value=self.initial["total_points_available"],
    step_size=self.initial["event"].step,
    label=""
)
```
and remove `points = forms.IntegerField(min_value=0, label="")` above? I think this would simplify things